cmd/web.go

```diff
@@ -71,6 +73,25 @@ func runHTTPRedirector() {
 	}
 }
 
+func runLetsEncrypt(listenAddr, domain, directory, email string, m http.Handler) error {
+	certManager := autocert.Manager{
+		Prompt:     autocert.AcceptTOS,
+		HostPolicy: autocert.HostWhitelist(domain),
+		Cache:      autocert.DirCache(directory),
+		Email:      email,
+	}
+	go http.ListenAndServe(":80", certManager.HTTPHandler(nil)) // all traffic coming into HTTP will be redirect to HTTPS automatically
```

Comment:
Shouldn't this also only listen on listenAddr ?

Reply:
for the HTTP validation that LetsEncrypt does, their servers need to be able to access port 80 on the server requesting the certificate.

Reply:
Maybe you should just write `:http`

Reply:
updated

Reply:
@flufmonster yes but listening only on listenAddr+":80" should be enough since it will be the facing interface also for web access. If I remember listenAddr is by default 0.0.0.0 so by default we have the same comportment but if someone want to listen only to one interface via configuring listenAddr we shouldn't listen on all interface even for LE.

Reply:
Thanks @sapk I've updated this so it only listens on listenAddr instead of 0.0.0.0

Reply:
I think this should be handled in same code as http to https functionality already built into gitea